src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs

```diff
@@ -882,39 +878,5 @@ internal override async Task<Solution> TryAddUsingsOrImportToDocumentAsync(Solut
             property = CreatePropertySymbol(propertyName, propertyType);
             return property != null;
         }
-
-        internal override IMethodSymbol GetDelegatingConstructor(
-            SemanticDocument document,
-            ObjectCreationExpressionSyntax objectCreation,
-            INamedTypeSymbol namedType,
-            ISet<IMethodSymbol> candidates,
-            CancellationToken cancellationToken)
-        {
-            var model = document.SemanticModel;
-
-            var oldNode = objectCreation
-                    .AncestorsAndSelf(ascendOutOfTrivia: false)
-                    .Where(node => SpeculationAnalyzer.CanSpeculateOnNode(node))
-                    .LastOrDefault();
-
-            var typeNameToReplace = objectCreation.Type;
-            var newTypeName = namedType.GenerateTypeSyntax();
-            var newObjectCreation = objectCreation.WithType(newTypeName).WithAdditionalAnnotations(s_annotation);
-            var newNode = oldNode.ReplaceNode(objectCreation, newObjectCreation);
-
-            var speculativeModel = SpeculationAnalyzer.CreateSpeculativeSemanticModelForNode(oldNode, newNode, model);
```

Comment:
no more speculative code.

Reply:
this was used so we could try to find good types for something like `new Foo(x => x.Length)`.  However, this is unnecessary as the semantic model already exposes helpers to ask stuff like "does `x => x.Length` convert to `Func<string, int>`?".

Reply:
Worth adding a test for that case, since it was so hidden?

Reply:
yes, that would be a good idea.